pycolleff/pycolleff/longitudinal_equilibrium.py

```diff
@@ -1121,24 +1180,358 @@ def calc_mode_coupling(
+        """."""
+        omegaps = (ps * h + cb_mode) * w0
+        hmps = _np.zeros((ms.size, ps.size, len(z_ij)), dtype=complex)
+        for iz, z in enumerate(z_ij):
+            hmps[:, :, iz] = LongitudinalEquilibrium.hmp(z, ms, omegaps)
+        return hmps
+
+    def lebedev_matrix(
+        self,
+        big_omega,
+        hmps,
+        ms,
+        ps,
+        cb_mode,
+        reduced=False,
+        adsyncfreq=True,
+        effsyncfreq="center",
+    ):
+        """Lebedev matrix to find root of determinant.
+
+        Applying sum over m in Eq. (18) of Ref. [2] results in
+        Eq. (33) and (34) of Ref. [3].
+        """
+        eqinfo = self.equilibrium_info
+        ring = self.ring
+        w0 = ring.rev_ang_freq
+        h = ring.harm_num
+        I0 = ring.total_current
+        E0 = ring.energy
+        C0 = ring.circum
+        alpha = ring.mom_comp
+        sigmae2 = ring.espread**2
+
+        psi_J = eqinfo["action_distribution"]
+        ws_J = _2PI * eqinfo["sync_freq"]
+        J = eqinfo["action"]
+
+        nr_ps = ps.size
+        if adsyncfreq:
+            dpsi_dJ = -ws_J * psi_J / (alpha * sigmae2 * _c)
+
+            c_omega = big_omega[0] + 1j * big_omega[1]
+
+            B_pp = _np.zeros((nr_ps, nr_ps), dtype=complex)
+
+            if reduced:
+                if _np.any(ms < 0):
+                    raise ValueError("reduced=True but m < 0 identified")
+                m2wJ2 = (ms[:, None] * ws_J) ** 2
+                m2wJ = ms[:, None] ** 2 * ws_J
+                mdpsi_dJ_div = 2 * m2wJ * dpsi_dJ / (c_omega * c_omega - m2wJ2)
+            else:
+                mdpsi_dJ_div = (
+                    ms[:, None] * dpsi_dJ / (c_omega - ms[:, None] * ws_J)
+                )
+
+            omegapp = (ps * h + cb_mode) * w0
+
+            idx_close = _np.zeros_like(ws_J)
+            # if ws_J.min() < c_omega.real < ws_J.max():
+            #     # print('here')
+            #     # print(c_omega.imag)
+            #     if _np.abs(c_omega.imag) < 1e-5:
+            #         idx_close = _np.isclose(c_omega.real, ws_J, rtol=1e-5)
+            #         print(_np.sum(idx_close))
+
+            # only resonators accept complex frequencies
+
+            zpp = self.get_impedance(w=omegapp + c_omega) / omegapp
+
+            # more general impedances
+            # zpp = self.get_impedance(w=omegapp + c_omega.real) / omegapp
+
+            # wavg = _2PI * eqinfo["avg_sync_freq"]
+            # zpp = self.get_impedance(w=omegapp + wavg) / omegapp
+
+            dpsi_dJ_ws = dpsi_dJ / ws_J
+
+            for ip in range(nr_ps):
+                h_mp = hmps[:, ip]
+                for ipp in range(nr_ps):
+                    h_mpp = hmps[:, ipp].conj()
+                    if _np.sum(idx_close):
+                        rgpp = (h_mp * h_mpp * dpsi_dJ_ws[None, :])[
+                            :, idx_close
+                        ]
+                        rgpp = rgpp.sum(axis=-1)
```

Comment:
since you have to break line 1284, consider doing:
```suggestion
                        rgpp = (h_mp * h_mpp * dpsi_dJ_ws[None, :])
                        rgpp = rgpp[:, idx_close].sum(axis=-1)
```

Reply:
Applied, thanks